assistants/prospector-assistant/assistant/agents/document_agent.py

```diff
@@ -83,13 +84,20 @@ async def draft_outline(
             context, config=config.agents_config.attachment_agent
         )
 
+        # get outline related info
+        outline: str | None = None
+        if path.exists(storage_directory_for_context(context) / "outline.txt"):
+            outline = (storage_directory_for_context(context) / "outline.txt").read_text()
```

Comment:
this is racy - the file could theoretically be deleted in between the existence check and the read.

a best practice is to try to read and catch the `FileNotFoundError`

Reply:
on a side note, the `Path` class has an `exists() -> bool:` function. so no need for `os.path`. as i said though, that's an antipattern and it's best to just try to read and handle the `FileNotFoundError`

Reply:
good idea. thanks for updating me.